### PR TITLE
Fixes #20403 - Only show authorized compute profiles in forms

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -492,10 +492,8 @@ def accessible_resource(obj, resource, order = :name, association: resource)
     list
   end
 
-  def accessible_related_resource(obj, relation, opts = {})
+  def accessible_related_resource(obj, relation, order: :name, where: nil)
     return [] if obj.blank?
-    order = opts.fetch(:order, :name)
-    where = opts.fetch(:where, nil)
```
Comment (tbrisker) [inline app/helpers/application_helper.rb:498]: not directly related to this but just noticed while i was working on this and cleaned up along the way.

File: app/models/compute_profile.rb [hunk 2]
```diff
@@ -20,5 +20,5 @@ class ComputeProfile < ApplicationRecord
   scoped_search :on => :name, :complete_value => true
   default_scope -> { order('compute_profiles.name') }
 
-  scope :visibles, -> { eager_load(:compute_attributes).where('compute_attributes.id > 0') }
+  scope :visibles, -> { joins(:compute_attributes) }
```
Comment (tbrisker) [inline app/models/compute_profile.rb]: we don't actually use the eager loaded attributes anywhere, this looks like some weird way to turn a left outer join into an inner join.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (dLobatog) [reply app/helpers/application_helper.rb:498]: Should be safe to do this https://github.com/search?q=accessible_related_resource&type=Code&utf8=%E2%9C%93

**Review by dLobatog — CHANGES_REQUESTED** [review]
@tbrisker Looks good, can you add a test to ensure this behavior remains? Just ensuring `visible_compute_profiles` does not return ComputeProfiles a user is unauthorized to see should be enough. Thanks!

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (tbrisker) [reply app/helpers/application_helper.rb:498]: not sure I understand your comment here, the behaviour doesn't change at all, just replace the ops hash into kwargs.

**Comment by tbrisker** [comment]
@dLobatog thanks, with the test i realized i had the wrong permission name there and that we don't have fixtures for compute profile permissions at all.

**Comment by tbrisker** [comment]
test failures seem to be unrelated.

**Review by boaz0 — APPROVED** [review]
LGTM

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @tbrisker !

**Pull request merged**